File: app/src/main/java/com/dani_chuks/andeladeveloper/starwars/data/db/DataSource.java
package com.dani_chuks.andeladeveloper.starwars.data.db;

import java.util.List;

import io.reactivex.Flowable;

public interface DataSource<T> {

    Flowable<List<T>> getAll();

    Flowable<List<T>> getItemsLimitedToSize(int size);

    Flowable<List<T>> getAllAlphabetically();

    Flowable<T> getItemByUrl(String stringUrl);

    void insertItem(T t);

    void insertItemList(List<T> t);

}

File: app/src/main/java/com/dani_chuks/andeladeveloper/starwars/data/db/DataSource.kt
package com.dani_chuks.andeladeveloper.starwars.data.db

import io.reactivex.Flowable

interface DataSource<T> {

    val all: Flowable<List<T>>

    val allAlphabetically: Flowable<List<T>>

    fun getItemsLimitedToSize(size: Int): Flowable<List<T>>

    fun getItemByUrl(stringUrl: String): Flowable<T>

    fun insertItem(t: T)

    fun insertItemList(t: List<T>)

}
